File: apps/zap-downloader/zap_downloader/workspace.py
import os
from pathlib import Path
from rich.console import Console

console = Console()

DEFAULT_WORKSPACE = "zap-workspace"


def get_workspace() -> str:
    return os.environ.get("ZAP_PACKAGES_WORKSPACE", DEFAULT_WORKSPACE)


def ensure_workspace(workspace_path: str = None) -> str:
    workspace = workspace_path or get_workspace()
    if not os.path.exists(workspace):
        os.makedirs(workspace, exist_ok=True)
        console.print(f"[green]Created workspace: {workspace}[/green]")
    return workspace

File: apps/zap-downloader-node/src/workspace.ts
import * as fs from 'fs';
import chalk from 'chalk';

const DEFAULT_WORKSPACE = 'zap-workspace';

export function getWorkspace(): string {
  return process.env.ZAP_PACKAGES_WORKSPACE || DEFAULT_WORKSPACE;
}

export function ensureWorkspace(workspacePath?: string): string {
  const workspace = workspacePath || getWorkspace();
  if (!fs.existsSync(workspace)) {
    fs.mkdirSync(workspace, { recursive: true });
    console.log(chalk.green(`Created workspace: ${workspace}`));
  }
  return workspace;
}
